docker/rootfs/etc/s6-overlay/s6-rc.d/frigate/finish

```diff
@@ -0,0 +1,30 @@
+#!/command/with-contenv bash
+# shellcheck shell=bash
+# Take down the S6 supervision tree when the service exits
+
+set -o errexit -o nounset -o pipefail
+
+declare exit_code_container
+exit_code_container=$(cat /run/s6-linux-init-container-results/exitcode)
+readonly exit_code_container
+readonly exit_code_service="${1}"
+readonly exit_code_signal="${2}"
+readonly service="Frigate"
+
+echo "Service ${service} exited with code ${exit_code_service} (by signal ${exit_code_signal})" >&2
+
+if [[ "${exit_code_service}" -eq 256 ]]; then
+  if [[ "${exit_code_container}" -eq 0 ]]; then
+    echo $((128 + exit_code_signal)) > /run/s6-linux-init-container-results/exitcode
+  fi
+elif [[ "${exit_code_service}" -ne 0 ]]; then
+  if [[ "${exit_code_container}" -eq 0 ]]; then
+    echo "${exit_code_service}" > /run/s6-linux-init-container-results/exitcode
+  fi
+else
+  # Exit code 0 is expected when Frigate is restarted by the user. In this case,
```

Comment:
So to be clear, if a user uses the WebUI to restart frigate, it will be restarted without the entire container restarting (assuming this is what you mean by go2rtc tolerating the restart: it will keep running)

Reply:
No. The old behavior still remains. This only hands the logic from Python to s6.

Whenever Frigate exits, s6 will exit the container as well. So from the Python side, all it is needed is to exit the Frigate process.

It is possible to achieve what you mentioned, but I don't think we should. It's not going to save much time anyway, because most of the time is spent during Frigate cleanup.

Reply:
agreed I was just making sure